### PR TITLE
New 404 page

### PR DESCRIPTION
**Pull request by lusol**
This 404 page appears every time you hit a dead link.

screen shot 2018-06-22 at 11 44 02 am
Gonna ask the design team to draw us some dope image and replace the above

related: 
https://github.com/screwdriver-cd/screwdriver/issues/769

### CODE REVIEW

File: app/components/404-display/template.hbs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class="test">
```
Comment (tkyi) [inline app/components/404-display/template.hbs]: why class "test"?

File: app/pipeline/route.js [hunk 2]
```diff
@@ -11,8 +11,8 @@ export default Route.extend(AuthenticatedRouteMixin, {
   },
 
   actions: {
-    error(reason) {
-      this.transitionTo('page-not-found', { path: `pipelines/${get(this, 'pipelineId')}`, reason });
+    error() {
+      return this.transitionTo('/404');
```
Comment (minz1027) [inline app/pipeline/route.js]: Why all error goes to 404? There cannot be other err when it cannot load the pipeline model. Can we check the statusCode and do 404 accordingly?

 https://guides.emberjs.com/release/routing/loading-and-error-substates/

**Comment by tkyi** [comment]
Test failing
```
11:53:45   not ok 118 Chrome 67.0 - ESLint | app: pipeline/route.js
11:53:45       ---
11:53:45           actual: >
11:53:45               false
11:53:45           expected: >
11:53:45               true
11:53:45           stack: >
11:53:45                   at Object.<anonymous> (http://localhost:7357/assets/tests.js:1305:12)
11:53:45                   at runTest (http://localhost:7357/assets/test-support.js:4108:30)
11:53:45                   at Test.run (http://localhost:7357/assets/test-support.js:4094:6)
11:53:45                   at http://localhost:7357/assets/test-support.js:4306:12
11:53:45                   at Object.advance (http://localhost:7357/assets/test-support.js:3746:26)
11:53:45                   at begin (http://localhost:7357/assets/test-support.js:5695:20)
11:53:45           message: >
11:53:45               pipeline/route.js should pass ESLint
11:53:45               
11:53:45               14:5 - Expected to return a value at the end of method 'error'. (consistent-return)
11:53:45               15:7 - Expected space(s) after "if". (keyword-spacing)
11:53:45           Log: |
11:53:45       ...
```

File: app/pipeline/route.js [hunk 3]
```diff
@@ -11,8 +11,12 @@ export default Route.extend(AuthenticatedRouteMixin, {
...
 
   actions: {
-    error(reason) {
-      this.transitionTo('page-not-found', { path: `pipelines/${get(this, 'pipelineId')}`, reason });
+    error(error) {
+      if (error.errors[0].status === 404) {
+        this.transitionTo('/404');
+      }
+
+      return true;
```
Comment (tkyi) [inline app/pipeline/route.js]: Why is this returning true?

Comment (lusol) [reply app/pipeline/route.js]: I dont think you can have an empty return value, so you need to return something.

File: app/pipeline/route.js [hunk 4]
```diff
@@ -11,8 +11,12 @@ export default Route.extend(AuthenticatedRouteMixin, {
   },
 
   actions: {
-    error(reason) {
-      this.transitionTo('page-not-found', { path: `pipelines/${get(this, 'pipelineId')}`, reason });
+    error(error) {
+      if (error.errors[0].status === 404) {
```
Comment (minz1027) [inline app/pipeline/route.js]: I think this should also be the case for template and command routes.

**Pull request merged**